### PR TITLE
Fixed LP Bug #1662059 by mentioning that JSON data type is supported …

### PR DESCRIPTION
**Pull request by borys-belinsky**
…for compressed columns and informing that generated columns can also be compressed by using a syntax extension;

Fixed small grammar and style issues

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
The style changes would be relevant for 5.6 too?

File: doc/source/flexibility/compressed_columns.rst [hunk 1]
```diff
@@ -5,54 +5,69 @@ Compressed columns with dictionaries
...
-* or a table can be modified to compress/decompress a column later:
+  
+* when altering a table and modifying a column to the compressed format:
   ``ALTER TABLE ... MODIFY [COLUMN] ... COLUMN_FORMAT COMPRESSED``, or
   ``ALTER TABLE ... CHANGE [COLUMN] ... COLUMN_FORMAT COMPRESSED``.
 
-To decompress a column, a ``COLUMN_FORMAT`` value any other than ``COMPRESSED``
-can be specified: ``FIXED``, ``DYNAMIC``, or ``DEFAULT``. If there is a column
+Unlike Oracle MySQL, compression is applicable to generated columns, both
+virtual and stored. Use this syntax extension as follows:
```
Comment (percona-ysorokin) [inline doc/source/flexibility/compressed_columns.rst]: is applicable to generated STORED columns only (it does not make sense for generated VIRTUAL columns as their values are calculated on the fly)

File: doc/source/flexibility/compressed_columns.rst [hunk 2]
```diff
@@ -5,54 +5,69 @@ Compressed columns with dictionaries
...
 ==============
 
-The feature is limited to InnoDB/XtraDB storage engine and to:
+The feature is limited to InnoDB/XtraDB storage engine and to columns of the
+following data types:
+
+* ``BLOB`` (including ``TINYBLOB``, ``MEDIUMBLOB``, ``LONGBLOG``)
 
-* ``BLOB`` (including ``TINYBLOB``, ``MEDIUMBLOB``, ``LONGBLOG``),
+* ``TEXT`` (including ``LONGTEXT``)
```
Comment (percona-ysorokin) [inline doc/source/flexibility/compressed_columns.rst]: (including TINYTEXT, MEDUUMTEXT, LONGTEXT)

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by borys-belinsky** [comment]
There is no style change. It is only correction of grammar and style which does not introduce new or different meaning. For this reason, I would suggest that other version be not changed.

**Pull request merged**